src/Components/Aspire.Elastic.Clients.Elasticsearch/AspireElasticClientsElasticsearchExtensions.cs

```diff
@@ -0,0 +1,153 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire;
+using Aspire.Elastic.Clients.Elasticsearch;
+using Elastic.Clients.Elasticsearch;
+using Elastic.Transport;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Extensions.Hosting;
+
+/// <summary>
+/// Extension methods for connecting Elasticsearch with Elastic.Clients.Elasticsearch client.
+/// </summary>
+public static class AspireElasticClientsElasticsearchExtensions
+{
+    private const string DefaultConfigSectionName = "Aspire:Elastic:Clients:Elasticsearch";
+    private const string ActivityNameSource = "Elastic.Transport";
+    /// <summary>
+    /// Registers <see cref="ElasticsearchClient"/> instance for connecting to Elasticsearch with Elastic.Clients.Elasticsearch client.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">A name used to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureSettings">An optional delegate that can be used for customizing options. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureClientSettings">An optional delegate that can be used for customizing ElasticsearchClientSettings.</param>
+    /// <exception cref="InvalidOperationException">Thrown when mandatory <see cref="ElasticClientsElasticsearchSettings.ConnectionString"/> is not provided.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when mandatory <see cref="ElasticClientsElasticsearchSettings.Cloud"/> is not provided and <see cref="ElasticClientsElasticsearchSettings.UseCloud" /> is true .</exception>
+    public static void AddElasticClientsElasticsearch(
```

Comment:
```suggestion
    public static void AddElasticsearchClient(
```

This reads better and follows our naming pattern used elsewhere.